lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -2937,4 +2940,87 @@ llvm::Expected<lldb::addr_t> GetTaskAddrFromThreadLocalStorage(Thread &thread) {
   return task_addr;
 #endif
 }
+
+namespace {
+
+struct TaskStatusRecord {
+  Process *process;
+  addr_t addr;
+
+  operator bool() const { return addr && addr != LLDB_INVALID_ADDRESS; }
+
+  static constexpr offset_t FlagsOffset = 0x0;
+  static constexpr offset_t ParentOffset = 0x8;
+  static constexpr offset_t TaskNameOffset = 0x10;
+
+  enum Kind : uint64_t {
+    TaskName = 6,
+  };
+
+  uint64_t getKind(Status &status) {
+    if (status.Success())
+      return process->ReadUnsignedIntegerFromMemory(
+          addr + FlagsOffset, sizeof(uint64_t), UINT64_MAX, status);
+    return UINT64_MAX;
+  }
+
+  std::optional<std::string> getName(Status &status) {
+    if (getKind(status) != Kind::TaskName)
+      return {};
+
+    addr_t name_addr =
+        process->ReadPointerFromMemory(addr + TaskNameOffset, status);
+    if (!status.Success())
+      return {};
+
+    std::string name;
+    process->ReadCStringFromMemory(name_addr, name, status);
+    if (status.Success())
+      return name;
+
+    return {};
+  }
+
+  TaskStatusRecord getParent(Status &status) {
+    addr_t parent = LLDB_INVALID_ADDRESS;
+    if (*this && status.Success())
+      parent = process->ReadPointerFromMemory(addr + ParentOffset, status);
+    return {process, parent};
+  }
+};
+
+struct Task {
+  Process *process;
+  addr_t addr;
+
+  operator bool() const { return addr && addr != LLDB_INVALID_ADDRESS; }
+
+  static constexpr offset_t ActiveTaskStatusRecordOffset = 0x8 * 13;
+
+  TaskStatusRecord getActiveTaskStatusRecord(Status &status) {
+    addr_t status_record = LLDB_INVALID_ADDRESS;
+    if (status.Success())
+      status_record = process->ReadPointerFromMemory(
+          addr + ActiveTaskStatusRecordOffset, status);
+    return {process, status_record};
+  }
+};
+
+}; // namespace
+
+llvm::Expected<std::optional<std::string>> GetTaskName(lldb::addr_t task_addr,
```

Comment:
Does the optional add anything over just using an empty string for  "success, but no name"?

Reply:
My testing shows that I can provide the empty string as a task name, which should be distinguished from no task name.

Reply:
To elaborate, using `std::optional<std::string>` allows both a task name of the empty string (which lldb should show), and a task not having a name.